edk2toolext/os/UefiVariableSupportLib.py

```diff
@@ -0,0 +1,310 @@
+# @file
+#
+# Python lib to support Reading and writing UEFI variables from Windows/Linux
+#
+#
+# Copyright (c), Microsoft Corporation
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+# GetUefiAllVarNames linux implementation is based on information from
+# https://github.com/awslabs/python-uefivars/blob/main/pyuefivars/efivarfs.py
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+
+import os
+import uuid
+import struct
+import logging
+
+from ctypes import (
+    create_string_buffer
+)
+
+if os.name == 'nt':
```

Comment:
I see that you've already separated out the windows logic and os agnostic logic, which is awesome. I think, however, that since this only works on windows currently, just to add an else statement to this initial check (not the others) that will raise a `NotImplementedError` stating `os.name` is not currently supported for this module.

Thoughts?

Reply:
This should contain the logic to support both windows and Linux.

Its joining both the windows and linux implementations under a single class to try to simplify for the consumers to be able to add a single class for the functionality.

If you have alternative ideas of how to do this, I would appreciate any pointers.  

Reply:
I missed that it was support for both ~ It's fine the way it is then :)